mod-source-record-manager-server/src/main/java/org/folio/dao/JournalRecordDaoImpl.java

```diff
@@ -183,36 +187,47 @@ public Future<Void> saveBatch(Collection<JournalRecord> journalRecords, String t
     LOGGER.info("saveBatch:: Saving {} journal records", journalRecords.size());
 
     try {
-      JsonObject[] records = journalRecords.stream()
-        .map(r -> new JsonObject()
-          .put("id", r.getId())
-          .put("job_execution_id", r.getJobExecutionId() != null ? r.getJobExecutionId() : null)
-          .put("source_id", r.getSourceId() != null ? r.getSourceId() : null)
-          .put("entity_type", r.getEntityType() != null ? r.getEntityType().toString() : EMPTY)
-          .put("entity_id", r.getEntityId())
-          .put("entity_hrid", r.getEntityHrId() != null ? r.getEntityHrId() : EMPTY)
-          .put("action_type", r.getActionType().toString())
-          .put("action_status", r.getActionStatus().toString())
-          .put("action_date", r.getActionDate())
-          .put("source_record_order", r.getSourceRecordOrder())
-          .put("error", r.getError() != null ? r.getError() : EMPTY)
-          .put("title", r.getTitle())
-          .put("instance_id", r.getInstanceId())
-          .put("holdings_id", r.getHoldingsId())
-          .put("order_id", r.getOrderId())
-          .put("permanent_location_id", r.getPermanentLocationId())
-          .put("tenant_id", r.getTenantId()))
-        .toArray(JsonObject[]::new);
-
-      Tuple tuple = Tuple.tuple().addArrayOfJsonObject(records);
-      return pgClientFactory.createInstance(tenantId).execute("SELECT insert_journal_records($1::jsonb[])", tuple)
-        .map((Void) null);
+      List<Tuple> tupleList = journalRecords.stream()
+        .map(this::prepareInsertQueryParameters)
+        .toList();
+      String query = format(INSERT_SQL, convertToPsqlStandard(tenantId), JOURNAL_RECORDS_TABLE);
+      LOGGER.trace("saveBatch:: query = {}; tuples = {}", query, tupleList);
+
+      return executeWithRetry(query, tupleList, tenantId, MAX_RETRIES, INITIAL_RETRY_DELAY_MS)
+        .mapEmpty();
     } catch (Exception e) {
       LOGGER.warn("saveBatch:: Error saving journal records", e);
```

Comment:
I would use the ERROR logging level here.

Reply:
in DataImportJournalBatchConsumerVerticle we are already logging it with error level.